bool point0
bool point1
bool point2
bool point3
